08-generic_cache/01_naive/cache.go

```diff
@@ -0,0 +1,37 @@
+package cache
+
+// Cache is key, value storage.
+type Cache[K comparable, V any] struct {
+	data map[K]V
+}
+
+// New create a new Cache with an initialised data.
+func New[K comparable, V any]() Cache[K, V] {
+	return Cache[K, V]{
+		data: make(map[K]V),
+	}
+}
+
+// Read returns the associated value for a key,
+// or ErrNotFound if the key is absent.
+func (c *Cache[K, V]) Read(key K) (V, error) {
+	v, ok := c.data[key]
+	if !ok {
+		return v, ErrNotFound
+	}
+	return v, nil
+}
+
+// Upsert overrides the value for a given key.
+func (c *Cache[K, V]) Upsert(key K, value V) error {
+	c.data[key] = value
+
+	// Do not return an error for the moment,
```

Comment:
If `Upsert` might return an error, then `Delete` might return as well. Or alternatively, if delete doesn't need it, then Upsert shouldn't need it as well.

Either way, I would leave it out, unless there's a reason.

Reply:
Done.